sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -1016,5 +1029,14 @@ object TableFunctionRegistry {
 }
 
 trait ExpressionBuilder {
+  def functionSignatures: Seq[FunctionSignature] = Nil
```

Comment:
using null/Nil is generally not advisable because if the user forgets to check, it leads to exceptions. Maybe make this return `Option[Seq[FunctionSignature]]` instead with `None` as the default instead?

Reply:
+1